pkg/ddc/alluxio/utils.go

```diff
@@ -186,28 +186,71 @@ func isPortInUsed(port int, usedPorts []int) bool {
 	return false
 }
 
-func (e *AlluxioEngine) parseRuntimeImage() (image, tag string) {
-	var (
-		defaultImage = "registry.cn-huhehaote.aliyuncs.com/alluxio/alluxio"
-		defaultTag   = "2.3.0-SNAPSHOT-2c41226"
-	)
+func (e *AlluxioEngine) parseRuntimeImage(image string, tag string, imagePullPolicy string) (string, string, string) {
+	if len(imagePullPolicy) == 0 {
+		imagePullPolicy = "IfNotPresent"
+	}
+	runtimeImage := common.DEFAULT_ALLUXIO_RUNTIME_IMAGE
+	runtimeImageInfo := strings.Split(runtimeImage, ":")
 
-	image, tag = docker.GetImageRepoTagFromEnv(common.ALLUXIO_RUNTIME_IMAGE_ENV, defaultImage, defaultTag)
-	e.Log.Info("Set image", "image", image, "tag", tag)
+	if len(image) == 0 {
+		defaultImage := runtimeImageInfo[0]
+		image = docker.GetImageRepoFromEnv(common.ALLUXIO_RUNTIME_IMAGE_ENV, defaultImage)
+	}
 
-	return
+	if len(tag) == 0 {
+		defaultTag := runtimeImageInfo[1]
+		tag = docker.GetImageTagFromEnv(common.ALLUXIO_RUNTIME_IMAGE_ENV, defaultTag)
+	}
+
+	e.Log.Info("Set runtime image", "image", image, "tag", tag)
+
+	return image, tag, imagePullPolicy
 }
 
-func (e *AlluxioEngine) parseFuseImage() (image, tag string) {
-	var (
-		defaultImage = "registry.cn-huhehaote.aliyuncs.com/alluxio/alluxio-fuse"
-		defaultTag   = "2.3.0-SNAPSHOT-2c41226"
-	)
+func (e *AlluxioEngine) parseFuseImage(image string, tag string, imagePullPolicy string) (string, string, string) {
+	if len(imagePullPolicy) == 0 {
+		imagePullPolicy = "IfNotPresent"
+	}
+	fuseImage := common.DEFAULT_ALLUXIO_FUSE_IMAGE
+	fuseImageInfo := strings.Split(fuseImage, ":")
 
-	image, tag = docker.GetImageRepoTagFromEnv(common.ALLUXIO_FUSE_IMAGE_ENV, defaultImage, defaultTag)
-	e.Log.Info("Set image", "image", image, "tag", tag)
+	if len(image) == 0 {
+		defaultImage := fuseImageInfo[0]
+		image = docker.GetImageRepoFromEnv(common.ALLUXIO_FUSE_IMAGE_ENV, defaultImage)
+	}
 
-	return
+	if len(tag) == 0 {
+		defaultTag := fuseImageInfo[1]
+		tag = docker.GetImageTagFromEnv(common.ALLUXIO_FUSE_IMAGE_ENV, defaultTag)
+	}
+
+	e.Log.Info("Set fuse image", "image", image, "tag", tag)
+
+	return image, tag, imagePullPolicy
+}
+
+func (e *AlluxioEngine) parseInitImage(image string, tag string, imagePullPolicy string) (string, string, string) {
```

Comment:
how about unify e.parseInitImage and r.parseInitImage?
I think it should be an until function because more Pods perhaps need init Container.

Reply:
I agree with you. I used to have the same idea, I will change the codes. Thanks.